Gem/Code/Source/GameLift/GameLiftClientComponent.cpp

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/Uuid.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Session/SessionConfig.h>
+
+#include "GameLiftClientComponent.h"
+#include <AWSGameLiftPlayer.h>
+#include <Request/AWSGameLiftAcceptMatchRequest.h>
+#include <Request/AWSGameLiftCreateSessionOnQueueRequest.h>
+#include <Request/AWSGameLiftCreateSessionRequest.h>
+#include <Request/AWSGameLiftJoinSessionRequest.h>
+#include <Request/AWSGameLiftSearchSessionsRequest.h>
+#include <Request/AWSGameLiftStartMatchmakingRequest.h>
+#include <Request/AWSGameLiftStopMatchmakingRequest.h>
+#include <ResourceMapping/AWSResourceMappingBus.h>
+
+namespace MultiplayerSample
+{
+    AZ_CVAR(bool, cl_joinaftersearch, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate,
+        "Whether to join the first available session after search automatically");
+    AZ_CVAR(bool, cl_acceptmatch, false, nullptr, AZ::ConsoleFunctorFlags::DontReplicate,
+        "Whether to accept the match automatically");
+
+    void GameLiftClientSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<GameLiftClientSystemComponent, AZ::Component>()->Version(0);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<GameLiftClientSystemComponent>(
+                      "GameLiftClientSystemComponent", "System Component for the GameLift client")

```

Comment:
nit: Call out this component is the MPS GameLift client so you can easily tell GameLift gem vs project components ie MPSGameLiftClientSystemComponent.